src/category_theory/monoidal/center.lean

```diff
@@ -117,19 +118,21 @@ def tensor_obj (X Y : center C) : center C :=
     begin
       dsimp,
       simp only [comp_tensor_id, id_tensor_comp, category.assoc, half_braiding.monoidal],
-      rw [pentagon_assoc, pentagon_inv_assoc, iso.eq_inv_comp, ←pentagon_assoc,
-        ←id_tensor_comp_assoc, iso.hom_inv_id, tensor_id, category.id_comp,
-        ←associator_naturality_assoc, cancel_epi, cancel_epi,
-        ←associator_inv_naturality_assoc (X.2.β U).hom,
-        associator_inv_naturality_assoc _ _ (Y.2.β U').hom, tensor_id, tensor_id,
-        id_tensor_comp_tensor_id_assoc, associator_naturality_assoc (X.2.β U).hom,
-        ←associator_naturality_assoc _ _ (Y.2.β U').hom, tensor_id, tensor_id,
-        tensor_id_comp_id_tensor_assoc, ←id_tensor_comp_tensor_id, tensor_id, category.comp_id,
-        ←is_iso.inv_comp_eq, inv_tensor, is_iso.inv_id, is_iso.iso.inv_inv, pentagon_assoc,
-        iso.hom_inv_id_assoc, cancel_epi, cancel_epi, ←is_iso.inv_comp_eq, is_iso.iso.inv_hom,
-        ←pentagon_inv_assoc, ←comp_tensor_id_assoc, iso.inv_hom_id, tensor_id, category.id_comp,
-        ←associator_inv_naturality_assoc, cancel_epi, cancel_epi, ←is_iso.inv_comp_eq, inv_tensor,
-        is_iso.iso.inv_hom, is_iso.inv_id, pentagon_inv_assoc, iso.inv_hom_id, category.comp_id],
+      -- On the RHS, we'd like to commute `((X.snd.β U).hom ⊗ 𝟙 Y.fst) ⊗ 𝟙 U'`
+      -- and `𝟙 U ⊗ 𝟙 X.fst ⊗ (Y.snd.β U').hom` past each other,
+      -- but there are some associators we need to get out of the way first.
+      slice_rhs 6 8 { rw pentagon, },
+      slice_rhs 5 6 { rw associator_naturality, },
+      slice_rhs 7 8 { rw ←associator_naturality, },
+      slice_rhs 6 7 { rw [tensor_id, tensor_id, tensor_id_comp_id_tensor, ←id_tensor_comp_tensor_id,
+        ←tensor_id, ←tensor_id], },
+      -- Now insert associators as needed to make the four half-braidings look identical
+      slice_rhs 10 10 { rw ←associator_inv_conjugation, },
+      slice_rhs 7 7 { rw ←associator_inv_conjugation, },
+      slice_rhs 6 6 { rw ←associator_conjugation, },
+      slice_rhs 3 3 { rw ←associator_conjugation, },
```

Comment:
A question form someone who isn't very familiar with monoidal categories: is there any hope for a more powerful coherence tactic where this setup work is not necessary?

Reply:
Yes, I think that lots more is possible!

* bicategories as axiomatized by @yuma-mizuno are I think much better behaved from the point of view of short formalisations of "string diagram" proofs --- precisely because you can't put two morphisms at the same horizontal level
* tactics to translate between the monoidal category axiomatization and the bicategory axiomatization will be helpful
* I'm planning on writing a `suppress_associators` tactic, which will absorb as many structural morphisms as possible into the `monoidal_comp`
* we can also consider writing a tactic that lets you refer to two morphisms in an expression, and the tactic will then try to "bring them together", by squeezing out structural morphisms from "in between them". This needs further specification. :-)